packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -544,6 +604,14 @@ class CircularProgressIndicator extends ProgressIndicator {
   /// The width of the line used to draw the circle.
   final double strokeWidth;
 
+  /// The progress indicator's line ending: square or round.
```

Comment:
> square or round

The name/boolean/square-round is kind of confusing here.

I think StrokeCap is still just the straightforward solution.

Reply:
The issue with strokeCap, as I tried to explain before, is that butt/square are always wrong. If you want butt, indeterminate is wrong. If you want square, value is wrong. Would we have a fourth value (null?) that goes between butt and square? I think that's too much complexity for the user. Even introducing a new enum could be easier to understand.

Reply:
Is StrokeCap not properly documented? Again, these are decisions I do not think are fair for us to make for the developer. I think this should be StrokeCap.

Reply:
How do you see the API working? Because either value (square/butt) would be breaking. If they are overridden, they don't matter because they are the same value.

Reply:
How would it be breaking? Does exposing this value to developers break the default behavior?

Reply:
Can the value of CircularProgressIndicator.strokeCap be nullable? in which case it would choose this default behavior when the developer has not provided a value?

Reply:
I can, but just to double check, you acknowledge this might be error prone, because no one would want a value that is represented wrongly (https://github.com/flutter/flutter/issues/109137#issuecomment-1209391300). For the "different indeterminate animation" I think it is fine (but different enough than the official material animation).

Reply:
> you acknowledge this might be error prone

I don't. I only pointed out that this value can be exposed without introducing a breaking change and suggested a way to do so. :)

Reply:
Oh, I totally agree. The weird thing for me is that there would be 4 values and only 2 would be really "correct". So even though we want to give users options they will shoot themselves with 2 of them (which seem the correct thing, you have `circle`, you want... `square`, bang).

Reply:
If the progress value starts on -10 instead of 0, we have a "deterministic" problem. No one would want it to start before, and if they want, we could just expose that number. Same thing, if they want value: 40, they don't want value: 50 (else we would expose like an offset and let they play with it).

For indeterminate, it is less bad because the material animation would change (most of them wouldn't notice), so I believe, if we want the material animation to be customizable, there are other ways of doing this (like allowing to change the speed, curves and delay).

Reply:
Just my 2cts, I think it is fair to assume that a determinate `CircularIndicator` is "correct" if it starts right at 90 degrees and that it draws half of a circle when its value is 0.5. Both are not true when using `StrokeCap.square` (not to say it is not 'circular' because ‘StrokeCap.square’ really means there are two squares drawn at both ends of the arc, which might makes sense if you want to ‘connect’ lines and arcs).

Nonetheless, @Pinks point about the versatility of the public API is valid, especially considering @bernaferrari following argument: “one day they might add other strokeCap effects, it makes sense.” (maybe Impeller will).

Reply:
I doubt so because CSS is 20 years old and only supports the same 3 strokeCap. But I agree they could. I just don't know what it could be that would add benefit to the framework.